clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -514,61 +536,76 @@ private void assertNextHeartbeatTiming(long expectedTimeToNextHeartbeatMs) {
 
     @Test
     public void testHeartbeatState() {
+        CommitRequestManager commitRequestManager = mock(CommitRequestManager.class);
```

Comment:
I would expect we don't need anything related to the commitRequestManager, because the `HBManager` we're testing knows nothing about it. It's the `MembershipMgr` the one that commits offsets before revoking partitions. Could you try removing this and the expectation on ln 571?